mlir/lib/Dialect/Rock/utility/loweringUtils.cpp

```diff
@@ -668,10 +679,25 @@ FailureOr<int64_t> mlir::rock::getNumCU(Operation *op) {
   }
   IntegerAttr numCU = maybeNumCU.value();
   AmdArchInfo archInfo = rock::lookupArchInfo(arch);
-  if (numCU.getSInt() < archInfo.minNumCU) {
+  if (numCU.getValue().getSExtValue() < archInfo.minNumCU) {
     return op->emitError() << "num_cu=" << numCU
                            << " cannot be lower than arch minNumCU="
                            << archInfo.minNumCU;
   }
-  return numCU.getSInt();
+  return numCU.getValue().getSExtValue();
+}
+
+FailureOr<UnitAttr> mlir::rock::getReverseGrid(Operation *op) {
```

Comment:
`FailureOr<UnitAttr>` could just be written `LogicalResult` but I see the genericness point.

Reply:
I d like to keep the pattern if thats alright.

Reply:
Yeah, it's fine